site/blog/tutorials/iasql-on-ec2.mdx

```diff
@@ -0,0 +1,260 @@
+---
+slug: iasql-on-ec2
+title: Deploy IaSQL on EC2 using IaSQL
+authors: [mtp1376]
+---
+
+We're intending to publish a series of blog posts on deploying different services using IaSQL. IaSQL covers a wide range of AWS services, so there are lots of possibilities to explore. You can deploy containers to ECS, launch servers using EC2, and have CodePipelines that take the burden of building and deploying your applications.
+
+In this post we'll explore how to deploy IaSQL on EC2, using IaSQL!
+
+### Start IaSQL
+To start using IaSQL capabilities, we'll need to run a local version of IaSQL:
+```bash
+docker run -p 9876:9876 -p 5432:5432 --name iasql iasql/iasql
+```
+This will download the latest IaSQL image from DockerHub and run it. After it's ready, you can access the dashboard at the following URL:
+```
+http://localhost:9876/
+```
+<img src={"/img/dashboard.png"} alt={"IaSQL dashboard"} />
+
+Now you can connect your AWS account in the dashboard. This will create a new database for you which you can connect to it using any Postgres client, or run your queries on it through the dashboard.
+
+If you'd prefer to use a hosted version, and not to run a local version of IaSQL, you can visit [app.iasql.com](https://app.iasql.com/) to use our SaaS solution.
+
+### Deploy IaSQL on EC2 with IaSQL
+
+#### Overview
+Let's begin by taking an overview of what we're going to do to give you a general idea of the parts that are going to be deployed, before we move on to explore the specifics.
+
+To deploy IaSQL on EC2, we need to:
+
+- Launch an EC2 instance that will host IaSQL
+- Set the EC2 user data so that Docker will be installed upon instance launch, and the IaSQL container will be started
+- Import an SSH key pair into AWS to be able to access the EC2 instance afterwards
+- Set a security group for the instance to allow SSH, while disallowing unauthorized access to the dashboard
+
+<img src={"/img/iasql-on-ec2.png"} alt={"IaSQL dashboard"} />
```

Comment:
Will there be dark-mode variants for all of these images?

Reply:
I'll add them. It makes sense.